From 2f9c114e1fd7fc7aff2cb7fe58187c127a3d1c4f Mon Sep 17 00:00:00 2001
From: Timotheus Bachinger <Timotheus.bachinger@checkmk.com>
Date: Sat, 30 Nov 2024 03:43:30 +0100
Subject: [PATCH 1/1] Enable create_venv for non-interactive runs (#167)

When using `create_venv` in non-interactive docker container runs, we
would get:
```
tput: No value for $TERM and no -T specified
```
This sets `TERM` in case it isn't and enables such runs.
---
 uv/private/create_venv.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/uv/private/create_venv.sh b/uv/private/create_venv.sh
index 10edaf4..b88b2d4 100644
--- a/uv/private/create_venv.sh
+++ b/uv/private/create_venv.sh
@@ -8,6 +8,10 @@ REQUIREMENTS_TXT="{{requirements_txt}}"
 
 PYTHON="$(realpath "$RESOLVED_PYTHON")"
 
+if [ -z ${TERM} ] || [ ${TERM} == "dumb" ]; then
+   export TERM=xterm
+fi
+
 bold="$(tput bold)"
 normal="$(tput sgr0)"
 
-- 
2.43.0

